zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -1334,3 +1337,87 @@ impl<F: Field> AbsWordGadget<F> {
         &self.is_neg
     }
 }
+
+/// Gadget to verify the byte-size of a word, i.e. the minimum number of bytes
+/// it takes to represent the word.
+#[derive(Clone, Debug)]
+pub(crate) struct ByteSizeGadget<F> {
+    /// Array of indices from which only one will be turned on. The turned on
+    /// index is the index of the most significant non-zero byte in value.
+    most_significant_nonzero_byte_index: [Cell<F>; N_BYTES_WORD + 1],
+    /// The inverse of the most significant non-zero byte in value. The inverse
+    /// should exist if the byte-size is non-zero.
+    most_significant_nonzero_byte_inverse: Cell<F>,
```

Comment:
In case `byte_size > 0` then we assign the inverse of the most significant non-zero byte to `most_significant_nonzero_byte_inverse`.

And we do use it here:
https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/ffc2d570f6458aa9ebaa0d983834f6fbb2ada3ae/zkevm-circuits/src/evm_circuit/util/math_gadget.rs#L1369-L1376

Which verifies that the product of the most significant non-zero byte and its inverse is in fact equal to `1`

Reply:
The case `i == 0` corresponds to `byte_size == 0`, and in that case, since there is no most significant non-zero byte, we don't have an inverse for it.

Reply:
Okay, I understand what you mean. In that case, should I assign a `0` for the case where `byte_size == 0`?

Reply:
I'm not quite sure how I can remove that in the case of `ByteSizeGadget`. Because `i` is the index in the array, it is not an advice column.

Reply:
I didn't fully understand. Do you mean remove the i==0 condition? And simply revert back to what it was initially?